### DISCUSSION TITLE
Welcome to ursina Discussions!

### DISCUSSION THREAD

**Discussion by pokepetter** [pokepetter/ursina#90] [General]
## Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by JasonMerit** [comment]
Hi guys. I'm having a difficulty running ursina from PyCharm. Specifically, I'm getting a 'modulenotfounderror no module named' error when importing ursina. I went through the installment steps without any issues, so I really don't know what's wrong.

**Reply by JasonMerit** [reply]
ayyy, just fixed it. I had to go to settings and install the package: https://www.jetbrains.com/help/pycharm/installing-uninstalling-and-upgrading-packages.html

**Comment by husainb12** [comment]
I have installed Ursina Game engine using pip install ursina put when i run my code in visual code studio it shows an error as follows:
    from usrina import *
ModuleNotFoundError: No module named 'usrina'

**Reply by husainb12** [reply]
Sorry i Just realised i Was spelling ursina wrong

**Comment by Joymaker** [comment]
Hello. I'm in the early stages of my first Ursina project and I'm trying to learn how to use the mouse powerfully and effectively. I would like to be able to drag on left click and do something different on right click. I would like to be able to detect a hit on an opaque object located inside a rather transparent object. I would like to know (or calculate) where on the target object I have hit.

In the Ursina cheat sheet, I find an extensive list of properties and functions for mouse, without nearly enough text to tell me how to really use them. Below are some of my favorites. Is there more text somewhere else? Is there a good advanced mouse example somewhere? Or should I plan on reading the source code?

Favorite mystery items: 
delta_drag = Vec3(0,0,0) 
raycast = True 
collision = None 

input(key)   
update()   
find_collision()   
unhover_everything_not_hit()

**Reply by Joymaker** [reply]
I would also like to know how to read the mouse wheel.

**Reply by pokepetter** [reply]
For getting where on the target you clicked, you want to use `mouse.point` or `mouse.world_point`.

`delta_drag ` is the difference between the position of the mouse when pressing and releasing the left mouse button.
The rest you can ignore. Those functions are internal.

The key names for scroll up and scroll down are `'scroll up'` and `'scroll down'`. Use them in an input function.

**Comment by themanh01** [comment]
Hi. I'm doing my own project for assignment, it almost finished. But when i try to convert it to .exe with pyinstaller, run it and i get this error:

Traceback (most recent call last):
  File "main.py", line 6, in <module>
    app = Ursina()
  File "ursina\main.py", line 27, in __init__
  File "direct\showbase\ShowBase.py", line 339, in __init__
  File "direct\showbase\ShowBase.py", line 1021, in openDefaultWindow
  File "direct\showbase\ShowBase.py", line 1056, in openMainWindow
  File "direct\showbase\ShowBase.py", line 766, in openWindow
  File "direct\showbase\ShowBase.py", line 746, in <lambda>
  File "direct\showbase\ShowBase.py", line 818, in _doOpenWindow
  File "direct\showbase\ShowBase.py", line 647, in makeDefaultPipe
  File "direct\directnotify\Notifier.py", line 130, in error
Exception: No graphics pipe is available!
Your Config.prc file must name at least one valid panda display
library via load-display or aux-display.

I have no idea to fix it, please help!

**Comment by Ukacha21** [comment]
Do you plan on making ursina build work for Android packages too? Such as .apk

**Comment by mrharisk** [comment]
[93mwarning: missing texture: 'assets/car' [0m
[93mwarning: missing texture: 'assets/road' [0m
[93mwarning: missing texture: 'assets/enemy' [0m

Hello sir i am facing this error i don't know what to do?

**Reply by Ukacha21** [reply]
Try to specify the texture file precisely including it's extension with a relative path. 

eg:


terrain = Entity(
    model= "quad",
    texture="files/textures/terrain_texture.png"
    )

**Comment by KidsAdventure** [comment]
Is it possible to make a game for linux in ursina?

**Reply by RosalieNL** [reply]
It is possible as ursina uses python. python is either preinstalled or easy to download on linux. if you want to make it easy to access you could look for a python to appimage script

**Comment by Proguko** [comment]
It's possible to add transparency to Entities in Ursina?

**Reply by Ukacha21** [reply]
Yes, using the alpha parameter

**Comment by Joymaker** [comment]
Would someone be willing to look at my inventory problem really quick?  I reduced this code from a game I'm working on, to try to make it easy to scan.  CubiconGraphic, functions input() and build_game() contain the heart of the matter.

I started with a stock Inventory class.  I add the "cog" item, it works fine.  I add an item made by generating an Entity from 3 cubes, it isn't draggable.  I'm pointing the mouse at it but mouse.hovered_entity prints as None.  What do I need to do to these cube objects to make them findable?

[CubiconQuestion.zip](https://github.com/pokepetter/ursina/files/12111489/CubiconQuestion.zip)

**Comment by pokepetter** [comment]
Please stop asking questions here. Make a new thread or an Issue.

**Comment by ghost** [comment]
Does anyone know how to change the color of the pointer on ursina, because mine is fuchsia but I would like it white, thanks.

**Reply by Ukacha21** [reply]
If you mean cursor pointer, you can access the FirstPersonController code and change yourself
